packages/react-components/src/components/InputBoxComponent.tsx

```diff
@@ -201,3 +634,818 @@ export const InputBoxButton = (props: InputBoxButtonProps): JSX.Element => {
     </TooltipHost>
   );
 };
+
+/* @conditional-compile-remove(mention) */
+/**
+ * Find mention tag if selection is inside of it
+ *
+ * @private
+ */
+const findMentionTagForSelection = (tags: TagData[], selection: number): TagData | undefined => {
+  let mentionTag: TagData | undefined = undefined;
+  for (let i = 0; i < tags.length; i++) {
```

Comment:
It's always preferred to use 
tags.forEach if you are not changing the array structure when you are doing the loop

Reply:
Can you go explain the why here please?

Reply:
Ok, so for the purpose that you can't mess with the array directly.

Reply:
Added to refactor task

Reply:
@PorterNan In this case, we are using `break` to break out of the loop early. Foreach doesn't work with break statement. We can use `every` instead, but using a for loop because of break statements is also legitimate.